OpenRA.Mods.Common/Traits/Buildings/Refinery.cs

```diff
@@ -144,12 +144,10 @@ public void OnDock(Actor harv, DeliverResources dockOrder)
 		{
 			if (!preventDock)
 			{
-				dockOrder.Queue(self, new CallFunc(() => dockedHarv = harv, false));
-				dockOrder.Queue(self, DockSequence(harv, self));
-				dockOrder.Queue(self, new CallFunc(() => dockedHarv = null, false));
+				dockOrder.QueueChild(self, new CallFunc(() => dockedHarv = harv, false));
```

Comment:
Another nice followup PR will be to rework `IAcceptResources` to expose a method to return the `DockSequence` activity plus notifications for start/end dock, and then this child queuing can move inside `DeliverResources` where it belongs.

Do you have a list of specific points you intend on working on after this PR is merged?
We should create an issue with these so we don't forget anything before the next release, and perhaps divide the tasks over a couple of people.

Reply:
The only specific point right now that I intend to work on after this PR is a fix for harvesters getting stuck around blossom trees / ore pits. This seems to me the most pressing issue still remaining and I already have an idea for how to fix it. I could take on some minor refactors of `HarvestResources` while I'm at it.

Aside from that, I have some vague ideas for new features (refinery rally points, reintroducing refinery locking in a more user friendly way) and refactors, but nothing I want to commit myself to finishing (or even starting) before the next release.

Reply:
Ok. I'll try on the weekend to collate my list of future-harv-pr-thoughts from above into a new issue that others can then add to.